components/list/src/list_formatter.rs

```diff
@@ -5,13 +5,28 @@
 use crate::provider::*;
 use crate::ListLength;
 use core::fmt::{self, Write};
+use icu_locale_core::preferences::{define_options, define_preferences};
 use icu_provider::marker::ErasedMarker;
 use icu_provider::prelude::*;
 use writeable::*;
 
 #[cfg(doc)]
 extern crate writeable;
 
+define_preferences!(
+    /// The preferences for list formatting.
+    ListFormatterPreferences,
+    {}
+);
+define_options!(
+    /// The options for list formatting.
+    ListFormatterOptions,
+    {
+        /// Style
+        style: ListLength
+    }
+);
+
```

Comment:
Issue: I thought we agreed to not touch the `options` argument. It's fairly easy in this crate, but the options bags are more complicated in other crates like `datetime` and I really don't want to get in the business of generating them with a cross-crate macro. Preferences, yes, but not options.

(I have other comments regarding this, but they are moot if we revert `ListFormatterOptions`)

Reply:
Sounds good. I hand-rolled the options bag. I believe each constructor should take `prefs, options` pair.